skrub/dataframe/_namespace.py

```diff
@@ -0,0 +1,101 @@
+import sys
+from types import ModuleType
+
+import pandas as pd
+
+import skrub.dataframe._pandas as skrub_pd
+import skrub.dataframe._polars as skrub_pl
+from skrub.dataframe._types import DataFrameLike
+
+
+def is_pandas(dataframe: DataFrameLike) -> bool:
+    """Check whether the input is a Pandas dataframe.
+
+    Parameters
+    ----------
+    dataframe : DataFrameLike
+        The input dataframe
+
+    Returns
+    -------
+    is_pandas : bool
+        Whether the dataframe is a Pandas dataframe or not.
+    """
+    return isinstance(dataframe, pd.DataFrame)
+
+
+def is_polars(dataframe: DataFrameLike) -> bool:
+    """Check whether the input is a Polars dataframe or lazyframe.
+
+    Parameters
+    ----------
+    dataframe : DataFrameLike
+        The input dataframe
+
+    Returns
+    -------
+    is_polars : bool
+        Whether the dataframe is a Polars dataframe/lazyframe or not.
+    """
+    if "polars" not in sys.modules:
+        return False
+
+    import polars as pl
+
+    return isinstance(dataframe, (pl.DataFrame, pl.LazyFrame))
+
+
+def get_df_namespace(*dfs: list[DataFrameLike]) -> tuple[ModuleType, ModuleType]:
+    """Get the namespaces of dataframes.
+
+    Introspects dataframes and returns their skrub namespace object
+    ``skrub.dataframe._{pandas, polars}`` and the dataframe module
+    ``{polars, pandas}`` itself.
+
+    The dataframes passed in input need to come from the same module, otherwise a
+    ``TypeError`` will be raised.
+
+    The outputs of this function are denoted ``skrub_px`` and ``px`` in reference to
+    the array API, returning namespace (NumPy, PyTorch and CuPy) as ``nx``.
+    Since we deal with Polars (``pl``) and Pandas (``pd``), we use ``px``
+    as a variable name.
+
+    Parameters
+    ----------
+    dfs : list[DataFrameLike],
```

Comment:
Is it possible to allow also getting the namespace for only one dataframe?

Reply:
Yes, thanks